managed/services/grafana/client.go

```diff
@@ -182,19 +182,27 @@ func (r role) String() string {
 }
 
 // getUserID returns user ID from Grafana for given user
-func (c *Client) getUserID(ctx context.Context, authHeaders http.Header) (int, error) {
+func (c *Client) GetUserID(ctx context.Context) (int, error) {
+	userID := -1
+
+	authHeaders, err := c.authHeadersFromContext(ctx)
+	if err != nil {
+		return userID, err
```

Comment:
if the user ID can't be 0, then let's return 0 instead of -1
```suggestion
		return 0, err
```